### ISSUE TITLE
Problem with single file components

### ISSUE THREAD

**Issue by johannes-z** [vuejs/vue#1917]
TL;DR: see last sentence.

I started using vue back when it was at version 0.12 or so and built a test application. Back then I didn't use single file components. I had script sections in the html files that had logic in them. It was.. ok-ish to build an app like that, but the problem was, when it got more complex, I couldn't always remember where the code I wanted to change was, and frankly I shouldn't need to remember if you separate concerns correctly.

What I then did was separating the vue (viewmodel) code into their own files and putting them all in a file. Then I had an app.js and imported all these single files using gulp. It worked great, BUT the view (html and css) was still in the according html file. So I then knew where to find the code I wanted or needed to change, but then I had to look for the html file where I used the viewmodel. 

I now wanted to try a very simple test application using the guide on http://vuejs.org/guide/application.html to build single file components... but using webpack + vue-loader seemed very impractical to use and I still don't fully know what's going on there (might be me, but the documentation describes this process rather vaguely). 

So I basically am looking for an easy, fast, efficient and nice way of separating concerns whilst being able to locate viewmodels and views fastly inside my project.

**Comment by OEvgeny** [comment]
Use forum for questions next time.
Where is an [example with vue-loader](https://github.com/vuejs/vue-loader-example) to learn how to do it.

**Comment by johannes-z** [comment]
Woops, sorry, must have confused tabs.
Anyway, I've seen this example, but it's not very intuitive, is it. And it looks like you have a lot of overhead using this method when creating a new component (importing and registering the component, css).
